modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/dht/topology/RentingPartitionIsOwnedDuringEvictionTest.java

```diff
@@ -137,25 +172,45 @@ private void testOwnedAfterEviction(boolean persistence, int backups) throws Exc
                     ds.addData(key, key);
             }
 
+            GridCacheContext<Object, Object> ctx0 = g0.cachex(DEFAULT_CACHE_NAME).context();
+            GridDhtLocalPartition evicting = ctx0.topology().localPartition(p0);
+
+            if (reservation == 2) {
+                GridDhtPartitionsReservation grpR =
+                    new GridDhtPartitionsReservation(ctx0.topology().readyTopologyVersion(), ctx0, "TEST");
+
+                evicting.reserve();
+                assertTrue(grpR.register(Collections.singleton(evicting)));
+                evicting.release();
+
+                assertTrue(grpR.reserve());
+            }
+            else if (reservation == 1)
+                evicting.reserve();
+
             IgniteEx joining = startGrid(backups + 1);
 
             if (persistence)
                 resetBaselineTopology();
 
-            GridDhtLocalPartition evicting = g0.cachex(DEFAULT_CACHE_NAME).context().topology().localPartition(p0);
+            doSleep(500);
```

Comment:
Can we use `GridTestUtils.waitForCondition` instead of `doSleep`?
`
if (reservations > 0)
    boolean res = GridTestUtils.waitForCondition(() -> evicting.state() == OWNING)...);
`

Reply:
The sleep is here to wait a bit before checking for OWNING state. It is expected to not change, so waiting for condition is meaningless.